### PR TITLE
Fix setting cookie path and domain options, add test to verify.

### PR DESCRIPTION
**Pull request by jamesgroat**
Similar to #7 the Path and Domain cookie options were not being set on the cookie store nor set on the actual cookies.

Added test to verify that the options are correctly set and show up on the actual cookie.

### CODE REVIEW

File: csrf_test.go [hunk 1]
```diff
@@ -41,6 +43,49 @@ func TestProtect(t *testing.T) {
...
+		t.Fatalf("cookie does not respect Domain option: got %v", cookie)
+	}
+	if !strings.Contains(cookie, "Max-Age=173") {
+		t.Fatalf("cookie does not respect MaxAge option: got %v", cookie)
+	}
+
+	// check for expected expire including the subsequent second to avoid edge case
+	if !strings.Contains(cookie, fmt.Sprintf("Expires=%s", expectedExpires.UTC().Format(http.TimeFormat))) &&
+		!strings.Contains(cookie, fmt.Sprintf("Expires=%s", expectedExpires.Add(time.Second).UTC().Format(http.TimeFormat))) {
+		t.Fatalf("cookie does not respect MaxAge option, unexpected expires got %v", cookie)
```
Comment (elithrar) [inline csrf_test.go]: This is failing on Go 1.4 on Travis: https://travis-ci.org/gorilla/csrf/builds/80532496

```go
-- FAIL: TestCookieOptions-2 (0.00s)
	csrf_test.go:85: cookie does not respect MaxAge option, unexpected expires got nameoverride=MTQ0MjM1NTg3M3xJbnBEWTBKdWNDdFhPVXh6TjNJdlRtbG5lVkJFYmtWM1RHOVNXU3RRVlhWc1QzUmFaWGhGWlc5RE1YYzlJZ289fIt9OI9hBudZlZ-pQwxLG-2N3AjeOyR_GhqxJGi3n_kd; Path=/pathoverride; Domain=domainoverride; Expires=Tue, 15 Sep 2015 22:27:26 UTC; Max-Age=173
```

File: csrf_test.go [hunk 2]
```diff
@@ -41,6 +43,49 @@ func TestProtect(t *testing.T) {
...
+	p := Protect(testKey, CookieName("nameoverride"), Secure(false), HttpOnly(false), Path("/pathoverride"), Domain("domainoverride"), MaxAge(173))(s)
+	p.ServeHTTP(rr, r)
+
+	if rr.Header().Get("Set-Cookie") == "" {
+		t.Fatalf("cookie not set: got %q", rr.Header().Get("Set-Cookie"))
+	}
+
+	cookie := rr.Header().Get("Set-Cookie")
+	if strings.Contains(cookie, "HttpOnly") {
+		t.Fatalf("cookie does not respect HttpOnly option: got %v", cookie)
```
Comment (elithrar) [inline csrf_test.go]: Can you update this and subsequent tests to show `got %q want %q` to make it easier to debug & align with the other tests.

**Comment by elithrar** [comment]
Thanks for catching these @jamesgroat - appreciated. Only a couple of comments and the failing test to fix and I'll merge.

**Comment by jamesgroat** [comment]
I've addressed the comments.

**Comment by elithrar** [comment]
Great - thanks again.

**Pull request merged**